crates/ruff_notebook/src/schema.rs

```diff
@@ -153,12 +153,36 @@ pub struct CodeCell {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub id: Option<String>,
     /// Cell-level metadata.
-    pub metadata: Value,
+    pub metadata: CellMetadata,
     /// Execution, display, or stream outputs.
     pub outputs: Vec<Value>,
     pub source: SourceValue,
 }
 
+/// Cell-level metadata.
+#[skip_serializing_none]
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+pub struct CellMetadata {
+    /// VS Code specific cell metadata.
+    ///
+    /// This is [`Some`] only if the cell's preferred language is different from the notebook's
+    /// preferred language.
+    /// <https://github.com/microsoft/vscode/blob/e6c009a3d4ee60f352212b978934f52c4689fbd9/extensions/ipynb/src/serializers.ts#L117-L122>
+    pub vscode: Option<CodeCellMetadataVSCode>,
+    /// Catch-all for metadata that isn't required by Ruff.
+    #[serde(flatten)]
+    pub extra: BTreeMap<String, Value>,
```

Comment:
What's the reason for using a `BTreeMap` over a normal `HashMap`? Or should we even use an `IndexMap` to preserve the field ordering?

Reply:
Yeah, I don't think it's required to use `BTreeMap` as we would sort the keys before writing it back to the file.